torchvision/prototype/features/_feature.py

```diff
@@ -96,18 +102,20 @@ def __torch_function__(
         with DisableTorchFunction():
             output = func(*args, **kwargs or dict())
 
-        # The __torch_function__ protocol will invoke this method on all types involved in the computation by walking
-        # the MRO upwards. For example, `torch.Tensor(...).to(features.Image(...))` will invoke
-        # `features.Image.__torch_function__` first. The check below makes sure that we do not try to wrap in such a
-        # case.
-        if not isinstance(args[0], cls):
-            return output
+            # Apart from `func` needing to be an exception, we also require the primary operand, i.e. `args[0]`, to be
+            # an instance of the class that `__torch_function__` was invoked on. The __torch_function__ protocol will
+            # invoke this method on *all* types involved in the computation by walking the MRO upwards. For example,
+            # `torch.Tensor(...).to(features.Image(...))` will invoke `features.Image.__torch_function__` with
+            # `args = (torch.Tensor(), features.Image())` first. Without this guard, the original `torch.Tensor` would
+            # be wrapped into a `features.Image`.
+            if func in cls._NO_WRAPPING_EXCEPTIONS and isinstance(args[0], cls):
+                return cls._NO_WRAPPING_EXCEPTIONS[func](cls, args[0], output)  # type: ignore[no-any-return]
```

Comment:
Optional Nit: you can avoid the double lookup on the dictionary by using `.get(func, None)`.